source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -160,22 +159,25 @@ void AuthenticatorImpl::startVerify() {
   }
 
   // Check "exp" claim.
-  // We use the current system time and allow for up to 5 seconds of slack.
+  // We use the current system time and allow for a configurable amount of slack.
   // Consider the following case:
   // AWS ALB receives a request and determines that the existing access token is valid 
   // (for another 1 seconds), forwards the request to Istio Ingressgateway and subsequently 
   // to some pod with an envoy sidecar. Meanwhile, 0.1 seconds have passed and when envoy checks 
   // the token it finds that it has expired.
-  now = absl::ToUnixSeconds(absl::Now());
+  const uint64_t now = absl::ToUnixSeconds(absl::Now());
+  const int64_t nbf_slack = jwks_data_->getJwtProvider().nbf_slack();
```

Comment:
jwks_data_ is only assigned in line 186,  you have to move this code after that.